lib/importer/nci_thesaurus_mirror.rb

```diff
@@ -0,0 +1,63 @@
+module Importer
+  class NciThesaurusMirror
+    attr_reader :parser, :version
+
+    def initialize(path, version = Time.now.utc.iso8601)
+      @parser = Obo::Parser.new(path)
+      @version = version
+    end
+
+    def import
+      parser.elements.each do |elem|
+        if valid_entry?(elem)
+          create_object_from_entry(elem)
+        end
+      end
+    end
+
+    def valid_entry?(entry)
+      semantic_types = semantic_types(entry)
+      (entry['id'].present? && entry['name'].present? && entry.respond_to?(:name) && entry.name == 'Term' &&
+        (semantic_types & ['Pharmacologic Substance', 'Pharmacological Substance', 'Clinical Drug', 'Therapeutic or Preventive Procedure']).length > 0 &&
+        (semantic_types & ['Obsolete_Concept']).length == 0)
+    end
+
+    def semantic_types(entry)
+        entry['property_value'].select{|s| s.start_with? 'NCIT:P106'}.map{|t| t.rpartition(' ')[0].split(' ', 2)[1].gsub!('"', '')}
```

Comment:
This line looked a little complicated so I started trying to work through it piece by piece. It does seem to work as is, but I wonder if it would be a little more clear with a regex in this case. Up to you if you think its better or not. I tested the regex itself on the file so I think it works, but the rest isn't fully tested.

```
def semantic_types
  matcher = /^NCIT:P106 "(?<semantic_type>.+)"/
  entry['property_values'].map { |s| s.match(matcher) }.compact.map { |s| s[:semantic_type] }
end
```
`match(some_regex) returns a match object if it matched, `nil` otherwise, `compact` removes all the `nils` from the list, and then `match_object[name of capture group]` gets the value out. (You don't have to use named capture groups and then you can just do `match_object[0]` or whatever, but I like naming them.

Up to you though, either way, you probably want to avoid `gsub!` (and most `!`) methods in general as their return values can bit you sometimes. Things with `!` modify the value inline instead of returning a copy with the modifications done. The non-`!` versions will always return a copy of the original object with the changes done (or not if they didn't apply). The `!`  versions however will often return not the object, but some sort of status indicator. For instance `gsub!` returns the string if it made any changes, but `nil` if it didn't, which can be a little surprising.